### ISSUE TITLE
Support fixed-width unicode strings 

### ISSUE THREAD

**Issue by mattjala** [HDFGroup/hsds#270]
HSDS currently does not support these (see `hdf5dtype.py:617`)

**Comment by ajelenak** [comment]
Yes, string encoding and how many bytes are reserved for its storage are decoupled.

**Comment by mattjala** [comment]
> Are they supported in the library?

Yep, see [here](https://github.com/mattjala/vol-tests/commit/a57d0bd6eb91f30a920699b880e817e1ec1a7997) for an example of fixed-length unicode strings being used in datasets/attributes - the native VOL passes both of these tests.

**Comment by ajelenak** [comment]
The question may be more related to how h5py treats HDF5 strings where this combo is not really supported. Any fixed-length string is treated as `bytes` object, not Unicode string.

**Comment by jreadey** [comment]
A fixed width unicode would be utf-32, but like @ajelenak says, it's not explicitly supported by the library. (or HSDS).

**Comment by mattjala** [comment]
> A fixed width unicode would be utf-32, but like @ajelenak says, it's not explicitly supported by the library. (or HSDS).

I think there's a confusion in terminology here. What I'm requesting is not support for a unicode character encoding where each particular character has a fixed width in bytes (e.g. UTF-32), but support for string datatypes that have a fixed total length in bytes (fixed length strings) AND have the character set/encoding UTF-8 (where a particular character does not have a fixed number of bytes associated with it). 

I've updated the title of this issue to be more clear. The library does support fixed-length strings in UTF-8 (See the tests I linked above).

**Comment by mattjala** [comment]
Implemented in #278
